packages/js/src/utils/id-token.ts

```diff
@@ -0,0 +1,68 @@
+import * as s from 'superstruct';
+
+const fullfillBase64 = (input: string) => {
+  if (input.length === 2) {
+    return `${input}==`;
+  }
+
+  if (input.length === 3) {
+    return `${input}=`;
+  }
+
+  return input;
+};
+
+/**
+ * @link [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)
+ */
+const IdTokenClaimsSchema = s.type({
+  iss: s.string(),
+  sub: s.string(),
+  aud: s.string(),
+  exp: s.number(),
+  iat: s.number(),
+  auth_time: s.optional(s.number()),
+  nonce: s.optional(s.string()),
+  acr: s.optional(s.string()),
+  amr: s.optional(s.array(s.string())),
+  azp: s.optional(s.string()),
+  at_hash: s.optional(s.string()),
+  c_hash: s.optional(s.string()),
+});
+
+export type IdTokenClaims = s.Infer<typeof IdTokenClaimsSchema>;
+
+/**
+ * Decode ID Token from JWT, without verifying.
+ *
+ * Verifying JWT requires fetching public key first,
+ * this can not be done in a sync function,
+ * in some cases, verifying is not necessary.
+ * @param token JWT string
+ * @returns IDToken combined with JWT Claims.
+ */
+export const decodeIdToken = (token: string): IdTokenClaims => {
+  const payloadPart = token.split('.')[1];
+
+  if (!payloadPart) {
+    throw new Error('invalid token');
+  }
+
+  const payloadString = payloadPart.replace(/-/g, '+').replace(/_/g, '/');
+  const json = decodeURIComponent(
+    escape(Buffer.from(fullfillBase64(payloadString), 'base64').toString())
+  );
+
+  try {
+    // Use Superstruct to validate the json type
+    const idToken = JSON.parse(json) as IdTokenClaims;
```

Comment:
don't think we need `JSON.parse` when we already imported `superstruct` especially using `s.assert`

Reply:
also, be careful with `JSON.parse` since it's extremely slow

Reply:
`decodeIdToken` is used to decode an ID Token without verification.

Using SuperSturct.Struct to parse ID Token seems that it cannot do it without verification.

![image](https://user-images.githubusercontent.com/10594507/150062748-f2841d5d-6291-4f2c-9701-d7e2d3dfdfb8.png)